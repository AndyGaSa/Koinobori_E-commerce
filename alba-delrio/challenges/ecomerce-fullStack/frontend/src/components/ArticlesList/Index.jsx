import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import loadArticles from '../../Redux/actions/actionCreators';
import './ArticleList.scss';

export default function AtriclesList() {
  const articles = useSelector((store) => store.articles);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(loadArticles());
  }, []);

  return (

    <ul>
      {articles.map((article) => (
        <li className="listElement" key={article.name}>
          <img className="listElement__img" src={article.img} alt="Ã¡randanos" />
          <div className="data">
            <p data-testid={`article-name-${article .name}`}>
              {article.name}
            </p>
            <span>
              {article.price}
              {' '}
              {article.currency}
            </span>
          </div>
          <div className="listElement__button"><button type="button">+</button></div>
        </li>
      ))}
    </ul>

  );
}
