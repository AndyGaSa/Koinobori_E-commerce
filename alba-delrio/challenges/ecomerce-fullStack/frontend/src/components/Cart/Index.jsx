import React from 'react';

export default function Cart() {
  return (

    <ul>
      <li>Cart</li>
    </ul>

  );
}
