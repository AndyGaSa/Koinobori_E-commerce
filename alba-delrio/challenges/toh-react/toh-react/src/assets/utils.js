import heroes from './constants';

export function update(heroId, name) {
  const heroList = JSON.parse(localStorage.getItem('heroes', heroes));
  const newHero = {
    name
  };
  const actualHeroList = heroList
    .map((hero) => (hero.id === +heroId ? { ...hero, ...newHero } : hero));
  localStorage.setItem('heroes', JSON.stringify(actualHeroList));
}

export function deleteHero(id, setHeroes) {
  const heroList = JSON.parse(localStorage.getItem('heroes', heroes));
  const newHeroList = heroList.filter((hero) => hero.id !== id);
  localStorage.setItem('heroes', JSON.stringify(newHeroList));
  setHeroes(newHeroList);
}

export function createHero(name, setHeroes, addHero) {
  let heroList = JSON.parse(localStorage.getItem('heroes'));
  heroList = heroList.sort((a, b) => a.id - b.id);
  const lastId = heroList[heroList.length - 1].id + 1;
  const newHero = {
    id: lastId,
    name

  };

  heroList.push(newHero);
  localStorage.setItem('heroes', JSON.stringify(heroList));
  setHeroes(heroList);
  addHero({ name: '' });
}
