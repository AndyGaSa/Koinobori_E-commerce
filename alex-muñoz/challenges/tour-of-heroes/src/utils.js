// heroeStorage es el array tocho
// filtra de mi array ,los objetos que alguna de sus propiedades contenga lo te hayan pedido
/* const localHero = heroStorage;
const findHero = heroesStorage.filter((hero) => {
  JSON.stringify(hero).contains(inputValue);
}); */

// function for delete an hero
const deleteButton = document.getElementById('delete');
this.addEventListener('click', () => { console.log(`${this.id}`); });
function deleteHero() {
  return console.log('hola');
}

// necesito poder acceder al elemento padre para poder luego acceder al hijo que contiene el objeto, o acceder al hermano.
