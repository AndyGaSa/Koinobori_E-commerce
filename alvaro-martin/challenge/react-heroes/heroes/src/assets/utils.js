/* eslint-disable no-restricted-globals */
/* eslint-disable max-len */
const HeroList = JSON.parse(localStorage.getItem('heroes'));

export default function update(he, id) {
  const newHero = {
    name: he,
  };
  const actualHeroList = HeroList.map((hero) => (hero.id === id ? { ...hero, ...newHero } : hero));
  localStorage.setItem('heroes', JSON.stringify(actualHeroList));
}
