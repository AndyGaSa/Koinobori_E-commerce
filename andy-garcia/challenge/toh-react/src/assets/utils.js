let localHeroes = JSON.parse(localStorage.getItem('heroes'));
export function update(heroId, name) {
  const newHero = {
    name,
  };
  localHeroes = localHeroes.map((hero) => (hero.id === +heroId ? { ...hero, ...newHero } : hero));

  localStorage.setItem('heroes', JSON.stringify(localHeroes));
  window.location.reload();
}
export function deleteHeroes(heroId) {
  localHeroes = localHeroes.filter(({ id }) => id !== heroId);
  localStorage.setItem('heroes', JSON.stringify(localHeroes));
  window.location.reload();
}
export function addHero(heroName) {
  const newHero = {
    id: (localHeroes[localHeroes.length - 1].id) + 1,
    name: heroName,
  };
  localHeroes.push(newHero);
  localStorage.setItem('heroes', JSON.stringify(localHeroes));
  window.location.reload();
}
