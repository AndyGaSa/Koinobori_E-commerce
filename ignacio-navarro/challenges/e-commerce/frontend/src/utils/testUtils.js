/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from '../redux/store';

function render(
  ui,
  {
    preloadedState,
    store = configureStore(preloadedState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <BrowserRouter><Provider store={store}>{children}</Provider></BrowserRouter>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
