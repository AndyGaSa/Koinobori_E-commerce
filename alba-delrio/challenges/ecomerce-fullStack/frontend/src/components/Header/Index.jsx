import React from 'react';
import './Index.scss';

export default function Header() {
  return (
    <>
      <div>
        <h2>EcoCesta</h2>
      </div>

    </>
  );
}
