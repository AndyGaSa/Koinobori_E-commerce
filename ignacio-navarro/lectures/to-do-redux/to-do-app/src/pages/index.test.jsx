import React from 'react';
import { render } from '../utils/utils-test';
import Index from '.';

describe('input component', () => {
  let sut;

  test('Should match snapshot', () => {
    sut = render(
      <Index />,
    );

    expect(sut).toMatchSnapshot();
  });
});
