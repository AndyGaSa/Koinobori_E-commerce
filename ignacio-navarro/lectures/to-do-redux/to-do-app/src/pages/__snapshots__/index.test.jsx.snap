// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <h1>
          My ToDo List 
        </h1>
      </header>
      <main>
        <div
          class="ToDo__input-container"
        >
          <h2>
            ToDo list
          </h2>
          <input
            name="addTask"
            placeholder="Add Task"
            type="text"
            value=""
          />
          <input
            name="addTask"
            placeholder="Add Description"
            type="text"
            value=""
          />
          <button
            class="ToDo__button"
            data-testid="btn-create"
            type="button"
          >
            Add
          </button>
        </div>
        <div
          class="ToDo__cards-container"
        />
      </main>
      <footer>
        <span>
          Copyright © Ignacio Navarro
           
          2021
           
        </span>
      </footer>
    </div>
  </body>,
  "container": <div>
    <header>
      <h1>
        My ToDo List 
      </h1>
    </header>
    <main>
      <div
        class="ToDo__input-container"
      >
        <h2>
          ToDo list
        </h2>
        <input
          name="addTask"
          placeholder="Add Task"
          type="text"
          value=""
        />
        <input
          name="addTask"
          placeholder="Add Description"
          type="text"
          value=""
        />
        <button
          class="ToDo__button"
          data-testid="btn-create"
          type="button"
        >
          Add
        </button>
      </div>
      <div
        class="ToDo__cards-container"
      />
    </main>
    <footer>
      <span>
        Copyright © Ignacio Navarro
         
        2021
         
      </span>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
